test/webdriverio/test/test_setup.ts

```diff
@@ -431,3 +431,40 @@ export async function isDragging(
     return workspaceSvg.isDragging();
   });
 }
+
+/**
+ * Returns the result of the specificied action precondition.
+ *
+ * @param browser The active WebdriverIO Browser object.
+ * @param action The action to check the precondition for.
+ */
+export async function checkActionPrecondition(
+  browser: WebdriverIO.Browser,
+  action: string,
+): Promise<boolean> {
+  return await browser.execute((action) => {
+    const node = Blockly.getFocusManager().getFocusedNode();
+    let workspace;
+    if (node instanceof Blockly.BlockSvg) {
+      workspace = node.workspace as Blockly.WorkspaceSvg;
+    } else if (node instanceof Blockly.Workspace) {
+      workspace = node as Blockly.WorkspaceSvg;
+    } else if (node instanceof Blockly.Field) {
+      workspace = node.getSourceBlock()?.workspace as Blockly.WorkspaceSvg;
+    }
+
+    if (!workspace) {
+      throw new Error('Unable to derive workspace from focused node');
+    }
+    const actionItem = Blockly.ShortcutRegistry.registry.getRegistry()[action];
+    if (!actionItem || !actionItem.preconditionFn) {
+      throw new Error(
+        `No registered action or missing precondition: ${action}`,
+      );
+    }
+    return actionItem.preconditionFn(workspace, {
+      focusedNode:
+        Blockly.FocusManager.getFocusManager().getFocusedNode() ?? undefined,
```

Comment:
Can `node` from above be reused here? I think technically it's guaranteed to be defined at this point, too, but you could always amend the conditional on line 456 to check for node's truthiness to placate the compiler (if needed).